dune-tests/sleek/lists-lists1.t/lists1.slk

```diff
@@ -0,0 +1,11 @@
+checkentail x=[|2|] |- x=(2:::[||]).
+
+checkentail x=(2:::(3:::[||])) |- x=(2:::(3:::[||])).
+
+checkentail x=(2:::z) |- exists y: x=(2:::y).
+
+checkentail x=[||] |- len(x)=0.
+checkentail x=(2:::(3:::(4:::[||]))) |- len(x) = 3.
+checkentail x=2:::y & len(y) = len_y |- len(x) = 1 + len_y.
+
+checkentail z=[||] & y=[||]:::z |- len(y) = 1.
```

Comment:
Curious, what happens on type error? Maybe add a couple of negative examples.

Reply:
In theory, type errors shouldn't come from the newly added inference system, since it imposes the same constraints as `Typeinfer`. Would it be worth to add a final check if all deduced types are consistent, just in case?

In practice, it seems to handle all current tests + this one.

Reply:
> In theory, type errors shouldn't come from the newly added inference system, since it imposes the same constraints as Typeinfer

Does this mean if there's a type error, it would be surfaced by something other than your new inference, so a negative test wouldn't be covering it anyway?

I was more just genuinely curious how a type error would show up, e.g. as a failed entailment, crash, etc.

Reply:
An error is thrown from here: https://github.com/hipsleek/hipsleek/blob/ad69efeea0b69a1d782903642d8d7ca647915a4e/src/typeinfer.ml#L186-L188

When SLEEK is invoked as a program, this shows up as the entailment having an `EXCast` result. When called via the API, it's currently a crash.